homeassistant/components/sensor/darksky.py

```diff
@@ -97,6 +98,14 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         _LOGGER.error("Latitude or longitude not set in Home Assistant config")
         return False
 
+    # Check to make sure that forecast is in a valid range of 1 week
+    forecast = config.get(CONF_FORECAST)
```

Comment:
Add this to the `PLATFORM_SCHEMA` validation.  You can also validate the day range  with something like `vol.Range(min=1, max=7)`.